### ISSUE TITLE
Communication

### ISSUE THREAD

**Issue by sobolevn** [jazzband/roadies#22]
I have a strong feeling that we are missing a huge thing here. Communication is the key to a good community.

What are my suggestions?

- [x] Slack team (I have actually already created a team for the `jazzband`)
- [ ] Integrations for different services
- [ ] Add links and information to website, social accounts, emails, etc

Anything I am missing?

p.s. I think I need to set @jezdez as a slack-admin, send me a note, please.

**Comment by jezdez** [comment]
@sobolevn Thanks for opening the issue, this is an important topic that I've by design left open so far since the community was simply not there yet when Jazzband started.

That said next time you start something like this I'd appreciate a note before to be able to voice my concerns about using Slack. This is not only fair in the spirit of the Jazzband but key to doing my job as roadie. Since you've decided to just go ahead you've effectively excluded me from a discussion which doesn't bode well for me trusting you with a community tool like a communication channel.

So to be absolutely clear, managing the Jazzband is **not** open to everyone and I believe you've jumped the gun with the Slack team. Please add me to the team so I feel comfortable again to consider using Slack. Then we can talk.

**Comment by sobolevn** [comment]
@jezdez Sorry for the rush. A team creation was indeed too early, but it was just a click away. I have sent you an invite to the email listed on your website. I will set you as an owner the moment you accept it, even if you do have concerns about using `slack` as a platform. But, do you? Anyway it can be freely removed with no harm.

I've seen a lot of good `slack` communities out there (`elixir`, `ios`, `react` as brief examples). What are the other options?

**Comment by kevin-brown** [comment]
> What are the other options?

[Gitter](https://gitter.im/) is a common alternative. Though I would personally lean towards Slack or staying with IRC.

**Comment by jezdez** [comment]
This is currently ongoing..

**Comment by jleclanche** [comment]
I have had a lot of success using [gitter-irc-bot](https://github.com/finnp/gitter-irc-bot) to do a two-way mirror between gitter and irc. It's easy to set up.

**Comment by triat** [comment]
Hi, do we have any follow up on this discussion ? I see that slack was checked but not really approved, what's the status now ?

**Comment by antonagestam** [comment]
Maybe it would be a good idea to try and describe what's lacking and what value you think a different set of tools adds. For me chat apps only add noise and stress and I think embracing isn't a great idea.

I would much prefer a forum kind if thing where the history of discussions is more approachable.

**Comment by jezdez** [comment]
I'll have a follow up eventually yeah, been investigating more platforms that would work from the roadies side.

One thing is clear though, Slack is not the solution for this as it would increase the barrier of participation again and cost a lot of money per active participant for essential functionality.

**Comment by triat** [comment]
If you're looking only for a forum like system, maybe having an other repo for the members with proper tags could do the job? I don't exactly know what's the lack right now, as I just arrived.

From my point of view, it could be great to have somewhere where we can, as new member, come and ask for the beginner questions and maybe find someone who will introduce you a bit with the repos.

There is no skill requirement to join you, and especially for that, I think that having help at start would really improve the first impression and the quality of the first participations.

**Comment by IvanAnishchuk** [comment]
I'd probably prefer gitter all else being equal. Slack, on the other hand, is something I use on daily basis for several teams. And I don't really think there's anything wrong with irc, it's been around for quite a while and it works while all these new services come and go.

**Comment by jezdez** [comment]
@IvanAnishchuk Yeah, there is a sweet spot for chat systems and IRC has historically been a constant, especially in the FOSS world. That said, there is a reason why so many other chat systems like Slack showed up: the barrier of entry of IRC is high for people that rarely interact via messaging. I know that this may be a generational issue, but since a goal of Jazzband is to lower the barrier of entry for contributors I'm not convinced IRC is the way to succeed. The current number of people in the #jazzband channel on Freenode (as advertised on https://jazzband.co/about/contact) is 12 (out of >300 members) and seems to point in that direction at least.

**Comment by IvanAnishchuk** [comment]
Well, I, for one, don't usually chat (except for work) unless I have some specific topic in mind. I'd probably join the slack team because I already have the app installed everywhere and it doesn't require much additional efforts but I'd be one of those silent members who rarely write at all. I'm not sure how representative my example is but I'm pretty sure that more people doesn't necessarily mean more active discussions. Note that I'm not disagreeing, slack and co. does indeed lower the barriers, is more mobile-friendly, can be accessed in browser with no extra software, and otherwise probably will attract more people to join.

Probably gitter rather than slack, it seems to be better optimized for open communities like ours. Unless everyone decides that slack is what they already use and adding another team is simpler than another app (we could probably try and poll to be sure).

**Comment by jezdez** [comment]
@IvanAnishchuk Agreed, I'm not particularly tied to Slack TBH, so it may just be Gitter. I'm gonna see if the new [GitHub team discussions ](https://github.com/blog/2471-introducing-team-discussions) may be enough for async communication. There is one for the "members" team now: https://github.com/orgs/jazzband/teams/members/discussions

**Comment by jstockwin** [comment]
@jezdez So we recently experienced this issue with JSON Models, and I think in our case the GitHub team discussions are not really sufficient. Here's some context on the specific issue:

Two of us have just joined the existing project lead, so there are three of us now. We immediately found ourselves in need of a place to chat. Initially we just used email, and then arranged a video call, however this does not work in the long term.

We really need some form of ongoing communication. The IRC on the Jazzband contact page no longer seems to work (?), and so for now we've created a discord server for the specific project (jsonmodels) which will just be used for the three of us to communicate with each other for now.

That said, it would be nice if we didn't need to set up our own stuff, and it would be really cool to have one central place for the wider Jazzband community.

I don't personally mind too much which platform we end up using (although I appreciate it needs some thought).

Let me know if there's anything I can do to help move things along -> I'm happy to compile some list of features for each platform if that's helpful?

**Comment by jezdez** [comment]
> @jezdez So we recently experienced this issue with JSON Models, and I think in our case the GitHub team discussions are not really sufficient. Here's some context on the specific issue:
> 
> Two of us have just joined the existing project lead, so there are three of us now. We immediately found ourselves in need of a place to chat. Initially we just used email, and then arranged a video call, however this does not work in the long term.
> 
> We really need some form of ongoing communication. The IRC on the Jazzband contact page no longer seems to work (?), and so for now we've created a discord server for the specific project (jsonmodels) which will just be used for the three of us to communicate with each other for now.
> 
> That said, it would be nice if we didn't need to set up our own stuff, and it would be really cool to have one central place for the wider Jazzband community.
> 
> I don't personally mind too much which platform we end up using (although I appreciate it needs some thought).
> 
> Let me know if there's anything I can do to help move things along -> I'm happy to compile some list of features for each platform if that's helpful?

Thanks for sharing that experience, I'll take that under advisement for a future solution that we will need to find.

My current state of thoughts on this: I'm considering signing up Jazzband for Google Workspaces (for non-profits, more on that soon), which would include Google Meet (up to 100 people) and the rest of the Google tool suite. Would Google Meet have sufficed for your use case?

With regard to Discord, I've seen it being used for a number of communities but am seriously concerned about the cost of maintenance (mostly moderation) in the current state of Jazzband. If we would have a larger number of roadies that would become less of a problem, but it's a non-trivial time investement.

**Comment by jstockwin** [comment]
Thanks for the update @jezdez !

> Would Google Meet have sufficed for your use case?

Google Meet is what we used for the call anyways, just using the free one (which I think gives you an hour at the moment) was sufficient. A downside of a video call is there is no record (other than any notes we took), and everyone has to be present at a certain given time.

What I feel we'd still be lacking is some kind of quick text-based communication (which we're using Discord for until some other solution is reached). Some way to (a) have [text] discussions about various topics (although maybe this can be handled on GitHub issues), and also (b) some way to quickly ask "hey, is this issue being handled or should I take a look"? I guess a group hangouts (or chat or whatever it's called now) would work for that, though. In particular, we came up with a list of tasks on the call but didn't really sort out who is doing what, since that can be done on Discord (or any other instant messaging platform).

Regarding Discord and moderation, that makes sense. However, I'd add that moderation is still an issue with Google Workspaces, it's just most of the communication would be on video calls (via Meet) or in private hangouts chats (which would be impossible to moderate). It feels like under Google Workspaces the moderation problem is being solved by simply not allowing any kind of semi-public (i.e. JazzBand members only?) discussions to happen like would be possible on Discord (or similar alternative), but perhaps I am missing some Google-based solution here.

As mentioned in other issues, I'd be very happy to help out as a roadie, and can help with some moderation.

In my view, it would be cool to have a `jsonmodels` channel/room or whatever where the core contributors can discuss things, but other members could easily get involved if they wanted to - I feel that this is one of the core ideals of JazzBand. There are some things that are too small for a video call, and too small/organisational for a GitHub issue, and we should have some solution to this.

Anyway, that's my thoughts, and hopefully it also helps you understand the issue we faced further. Thanks for taking the time to think about this!

**Comment by beregond** [comment]
Hey @jezdez 

From my experience even if you have access to google workspaces (which would be great, anyway) - you are still using something other for communication, like slack (or, free option, discord).

I understand though the problem with moderation

**Comment by jezdez** [comment]
We now have project teams in Jazzband, so that allows those teams to use the GitHub team discussions, e.g. https://github.com/orgs/jazzband/teams/pip-tools.

**Comment by graingert** [comment]
Hello, now that gitter has been donated to the matrix project, its usability has increased hugely and I think it's worth another look

**Comment by graingert** [comment]
see also https://github.com/jazzband/help/issues/225 and https://github.com/jazzband/help/pull/236

**Comment by jezdez** [comment]
I wish you would not have done that, @graingert, since you're basically forcing me now to monitor this channel for CoC violations now without a plan or documentation what this channel is supposed to be used for. Please be so kind and delete the channel again or let Gitter staff know that it was created by accident.

**Comment by graingert** [comment]
> I wish you would not have done that, @graingert, since you're basically forcing me now to monitor this channel for CoC violations now without a plan or documentation what this channel is supposed to be used for.

I plan on bridging it per #225 so there's no extra code of conduct overhead

> Please be so kind and delete the channel again or let Gitter staff know that it was created by accident.

By accident?

**Comment by graingert** [comment]
I'm looking into deleting the channel, and I'm currently trying to contact you directly on IRC. I do think it would be a great shame to delete the gitter community before it has had a chance to flourish

**Comment by jezdez** [comment]
> > I wish you would not have done that, @graingert, since you're basically forcing me now to monitor this channel for CoC violations now without a plan or documentation what this channel is supposed to be used for.
> 
> I plan on bridging it per #225 so there's no extra code of conduct overhead

I haven't agreed to that ticket and haven't been presented with a reason why we should be on Gitter AND Libera (using both platforms, no matter the bridging). @RonnyPfannschmidt just opened it when the Freenode/Libera kerfuffle happened. I think you're assuming that it's okay to just add new services to Jazzband, but it's my experience to first really consider if it would actually be maintainable. That Gitter allows org members to create channels is a huge red flag for me for example.

> > Please be so kind and delete the channel again or let Gitter staff know that it was created by accident.
> 
> By accident?

I don't know what else to call your decision to just create a channel without coordinating with me.

**Comment by graingert** [comment]
> I'm currently trying to contact you directly on IRC

@jezdez and I had a meeting regarding these communication channels and here's a summary:

* **the "accident"**: I requested access to gitter via github apps and it was approved so I was under the impression I wasn't acting unilaterally.
* **the EoL of gitter.im**: gitter is two things now. a Synapse server and a https://gitlab.com/gitterHQ/webapp only the webapp is on "end of life care", and once element is feature equivalent with gitter.im the webapp will be replaced maintaining the same logs, URLs and theme.
* **the CoC moderation overhead**: bridging is truly seamless and so you could moderate from your @jezdez:matrix.org account and or @jezdes:libera.chat account as you do now already. You would just notice that @...:gitter.im users were arriving in the channel and you'd not have to resolve a username to a github account before banning a user
* **the maintenance overhead**: of course there will be netsplits! but resolving netsplits is the novel part of matrix: https://matrix.org/#:~:text=Matrix%20is%20really%20a%20decentralised%20conversation%20store%20rather%20than%20a%20messaging%20protocol. the gitter and matrix team have been very responsive in terms of fixing/resolving teething issues the Twisted team has had, eg coming into the rooms and poking them from both ends and monitoring and updating me on global federation outages

**Comment by Archmonger** [comment]
Since this ticket was first opened 6 years ago, GitHub now has a built in communications platform in the form of GitHub Discussions.

My personal opinion is the best path forward here is opening a discussion tab within either this repo, or a separate discussions repo. External platforms are a bit iffy since it requires users to follow Jazzband on a whole different platform.

There definitely is difficulty getting Jezdez's thoughts communicated across the group. As long as we have a central & convenient location for communication, this situation is improved considerably.

**Comment by webknjaz** [comment]
FTR there's a matrix space now: https://matrix.to/#/%23jazzband:matrix.org
